src/recon_rec.erl

```
@@ -80,25 +82,29 @@ list() ->
 -spec get_list() -> [listentry()].
 get_list() ->
     ensure_table_exists(),
-    Lst = lists:map(fun make_list_entry/1, ets:tab2list(ets_table_name())),
+    Lst = lists:map(fun make_list_entry/1, ets:tab2list(records_table_name())),
     lists:sort(Lst).
 
 %% @doc Limit output to selected fields of a record (can be 'none', 'all', a field or a list of fields).
 %% Limit set to 'none' means there is no limit, and all fields are displayed; limit 'all' means that
 %% all fields are squashed and only record name will be shown.
 %% @end
--spec limit(record_name(), arity(), limit()) -> ok | {error, record_unknown}.
-limit(Name, Arity, Limit) ->
+-spec limit(record_name(), arity(), limit()) -> ok | {error, any()}.
+limit(Name, Arity, Limit) when is_atom(Name), is_integer(Arity) ->
     case lookup_record(Name, Arity) of
         [] ->
             {error, record_unknown};
         [{Key, Fields, Mod, _}] ->
-            ets:insert(ets_table_name(), {Key, Fields, Mod, Limit}),
+            ets:insert(records_table_name(), {Key, Fields, Mod, Limit}),
             ok
-    end.
+    end;
+limit(_, _, _) ->
+    {error, "Bad argument - the spec is limit(atom(), integer(), limit())"}.
```

let it crash, document the fields instead.